File: ABC/<100/017/C++/B.cpp
#include<iostream>
using namespace std;


bool isChoku(string s){
  int n = s.size();
  s += 'x';
  for(int i = 0; i < n; ++i){
    if(s[i] == 'c'){
      ++i;
      if(s[i] == 'h') continue;
      return false;
    }
    if(s[i] == 'o') continue;
    if(s[i] == 'k') continue;
    if(s[i] == 'u') continue;

    return false;
  }

  return true;
}


int main(){
  string X; cin >> X;
  cout << (isChoku(X) ? "YES" : "NO") << endl;
  return 0;
}

File: ABC/<100/017/Haskell/B.hs
main :: IO ()
main = do
    x <- getLine
    putStrLn $ if isChoku x then "YES" else "NO"


isChoku :: String -> Bool
isChoku [] = True
isChoku ('c':'h':rest) = isChoku rest
isChoku ('o':rest)     = isChoku rest
isChoku ('k':rest)     = isChoku rest
isChoku ('u':rest)     = isChoku rest
isChoku _              = False
